src/MainWindow.vala

```diff
@@ -44,6 +44,8 @@ public class AppCenter.MainWindow : Gtk.ApplicationWindow {
     private int homepage_view_id;
     private int installed_view_id;
 
+    private static int valid_query_length = 3;
```

Comment:
use a const here instead of static, this requires a rename to uppercase "VALID_QUERY_LENGTH"